datafusion/src/logical_plan/expr.rs

```diff
@@ -1118,36 +1133,56 @@ pub fn columnize_expr(e: Expr, input_schema: &DFSchema) -> Expr {
     }
 }
 
+/// Recursively replace all Column expressions in a given expression tree with Column expressions
+/// provided by the hash map argument.
+pub fn replace_col(e: Expr, replace_map: &HashMap<&Column, &Column>) -> Result<Expr> {
```

Comment:
this is used in predicate push down optimizer to push predicates to both sides of the join clause.

Reply:
use `pub(crate)`?